lib/elasticsearch_query/filter_formatter/match.rb

```diff
@@ -2,7 +2,11 @@ module ElasticsearchQuery
   module FilterFormatter
     class Match < Base
       def to_hash
-        { match: { @name => value } }
+        if array?
+          { terms: { @name => value } }
+        else
+          { match: { @name => value } }
+        end
```

Comment:
I'm confused, your PR says that the end solution uses a match query but the code here says it uses a term.

Term queries are very different, it's much more like an "equals" predicate, because the terms are left un-analyzed.

Reply:
+1. term queries are quite different. I think we just need a match query that separates the multiple values by space.

Reply:
PR message was incorrect when copy and pasting.  it is supposed to be an `equals` predicate

Reply:
the space actually doesn't work, the doc reference uses `q` query for multi-word and scoring.  when injecting with a space, the query itself doesn't fail but it is actually checking for the exact phrase "a b" not "a" or "b"

Reply:
I am not sure about it. Look at the [match's documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html). Specifically at the operator. The default value is `OR`.

Separately - If you want to use the term, I think we need to use `terms` since it supports multiple values. https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html

I will ask this separately, how did we test this E2E?

Reply:
Created a new test case in the audits service as well as manually tested.

The proposed solution of match with a space can be achieved by testing `https://staging1a.outreach-staging.com/api/v2/audits?filter[eventName]=login_failure login_success`.  This results in 0 results when there are events with `login_failure`/`login_success` in staging.

Reply:
I see that you are using terms now. I think this should be okay. Thanks

Question about the match - Did you concatenate the comma-separated values while using match?

Reply:
The gem itself was already doing string split on `,` and producing an array